### PR TITLE
Isoltest SMTChecker option and BMC specific tests

### PR DESCRIPTION
**Pull request by leonardoalt**
Adds an isoltest option for the SMTChecker engine, which then allows new tests that were uncovered before, specific to the BMC engine

### CODE REVIEW

File: test/libsolidity/SMTCheckerTest.cpp [hunk 1]
```diff
@@ -47,14 +45,19 @@ SMTCheckerTest::SMTCheckerTest(string const& _filename): SyntaxTest(_filename, E
 	if (!available.cvc4)
 		m_enabledSolvers.cvc4 = false;
 
-	if (m_enabledSolvers.none())
+	auto engine = ModelCheckerEngine::fromString(m_reader.stringSetting("SMTEngine", "all"));
```
Comment (axic) [inline test/libsolidity/SMTCheckerTest.cpp:48]: What does this do on an invalid input? Shouldn't that be reported like above with the exception?

Comment (leonardoalt) [reply test/libsolidity/SMTCheckerTest.cpp:48]: Right, added

**Review by axic — DISMISSED** [review]

**Comment by leonardoalt** [comment]
Depends on https://github.com/ethereum/solc-js/pull/497

**Comment by leonardoalt** [comment]
@axic can you reapprove

**Review by axic — APPROVED** [review]

**Pull request merged**